.github/workflows/ci.yml

```diff
@@ -6,13 +6,14 @@ on:
   pull_request:
 jobs:
   unit:
-    name: ${{ matrix.os }} jdk-${{ matrix.java }} unit tests
+    name: ${{ matrix.os }} jdk-${{ matrix.java }} unit ${{matrix.type}} tests
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         os: [windows-latest, macOS-latest, ubuntu-latest]
         java: [8, 11]
+        type: [build, debug, extensions, navigation, refactor, tools, util]
```

Comment:
We should be able to speed up the CI by ~2x by only introducing two types instead of 7 types.

Reply:
I kind went overboard here :smile: 2 or 3 might be enough.